examples/video_player/src/lib.rs

```diff
@@ -0,0 +1,359 @@
+                render_play_controls(self, ctx, ui);
+                render_image(self, ui);
+            }
+        });
+    }
+}
+
+fn render_navbar(app: &mut MyApp, ui: &mut eframe::egui::Ui) {
+    ui.horizontal(|ui| {
+        // Clear Button
+        ui.add_enabled_ui(app.app_state != AppState::Stopped, |ui| {
+            if ui.button("clear").clicked() {
+                app.app_state = AppState::Stopped;
+                app.playback_speed = 1;
+            }
+        });
+
+        // Load Button
+        ui.add_enabled_ui(app.app_state == AppState::Stopped, |ui| {
+            if ui.button("load").clicked() {
+                log::info!("Loading the video, Path: {}", app.file_path);
+                let file_path = PathBuf::from(&app.file_path);
+                let mut video_reader = VideoReader::new(&file_path, ImageFormat::Rgb8).unwrap();
+                video_reader.start().unwrap();
+
+                app.app_state = AppState::Playing(video_reader, Instant::now(), None);
+            }
+
+            ui.add_sized(
+                [ui.available_width(), ui.available_height()],
+                TextEdit::singleline(&mut app.file_path).hint_text("Click to set path"),
+            )
+        });
+    });
+}
+
+// This function only runs if `VideoReader` is present
+fn render_play_controls(app: &mut MyApp, ctx: &eframe::egui::Context, ui: &mut eframe::egui::Ui) {
+    // Slider
+    let mut is_finished = false;
+    if let AppState::Playing(video_reader, ..) = &mut app.app_state {
+        if let Some(duration) = video_reader.get_duration() {
+            if let Some(current_pos) = video_reader.get_pos() {
+                app.slider_pos = current_pos.as_secs();
+
+                // Video is now finished
+                if current_pos == duration {
+                    is_finished = true;
+                }
+            }
+
+            if ui
+                .add_sized(
+                    egui::Vec2::new(ui.available_width(), SLIDER_HEIGHT),
+                    egui::Slider::new(&mut app.slider_pos, 0..=duration.as_secs()),
+                )
+                .interact(egui::Sense::DRAG)
+                .drag_stopped()
+            {
+                let seek_time = std::time::Duration::from_secs(app.slider_pos);
+
+                #[allow(unused_must_use)]
+                video_reader.seek(SeekFlags::TRICKMODE, seek_time);
+            }
+        }
+    }
+
+    if is_finished {
+        let app_state = match std::mem::take(&mut app.app_state) {
+            AppState::Playing(video_player, ..) => AppState::Finished(video_player),
+            _ => panic!("Unexpected AppState"),
+        };
+        app.app_state = app_state;
+    }
+
+    // egui Windows
+    // Control Window
+    egui::Window::new("Control").show(ctx, |ui| {
+        ui.horizontal(|ui| {
+            Grid::new("control_grid").show(ui, |ui| {
+                let button_text = match app.app_state {
+                    AppState::Playing(..) => "Stop",
```

Comment:
```suggestion
                    AppState::Playing(..) => "Pause",
```
it's a bit confusing the naming